Model/Api/OrderManagement.php

```diff
@@ -133,10 +133,14 @@ public function manage(
 
             $this->logHelper->addInfoLog($this->request->getContent());
 
-            $orderStoreId = $this->orderHelper->getOrderStoreIdByDisplayId($display_id);
-            $this->hookHelper->setMagentoStoreId($orderStoreId);
+            // get the store id. try fetching from quote first,
+            // otherwise load it from order if there's one created
+            $storeId = $this->orderHelper->getStoreIdByQuoteId($order)
+                    ?: $this->orderHelper->getOrderStoreIdByDisplayId($display_id);
 
-            $this->logHelper->addInfoLog('Order StoreId: ' . $orderStoreId);
+            $this->hookHelper->setMagentoStoreId($storeId);
+
+            $this->logHelper->addInfoLog('StoreId: ' . $storeId);
 
             $this->hookHelper->setCommonMetaData();
             $this->hookHelper->setHeaders();
```

Comment:
We have special method for checking webhook.
Instead of 
```
  $this->hookHelper->setMagentoStoreId($storeId);
  $this->hookHelper->setCommonMetaData(); 
  $this->hookHelper->setHeaders();
  $this->hookHelper->verifyWebhook();
```

we can simply use

```
  $this->hookHelper->preProcessWebhook($storeId)
```

Reply:
correct